src/select.coffee

```diff
@@ -341,7 +342,15 @@ Ember.AddeparMixins.ResizeHandlerMixin,
 
   keyDown: (event) ->
     # show dropdown if dropdown is not already showing
-    return @set('showDropdown', yes) unless @get 'showDropdown'
+    # and the keycode should be one of special keys or common characters
+    # 'A'..'Z', 'a..z','0..9'
+    if event.keyCode in [@KEY_CODES.ENTER, @KEY_CODES.SPACEBAR] or
+    event.keyCode in [@KEY_CODES.DOWN, @KEY_CODES.UP] or
```

Comment:
for this specific case, i'd make a static map with all the keys that match, and then check that map.

for general cases, when you have 40 different clauses in your `if` statement, its better to put each clause into a variable name that makes sense and then use those names, so that the statement is more readable.